pdf_annotate/util/geometry.py

```diff
@@ -76,6 +76,26 @@ def matrix_multiply(*args):
     return r
 
 
+def matrix_inverse(matrix):
+    """Invert a 6-item homogenous transform matrix.
+
+    A transform matrix [a, b, c, d, e, f] is part of a 3x3 matrix:
+
+        a b 0
+        c d 0
+        e f 1
+
+    Invert it using the formula for the inverse of a 3x3 matrix from
+    http://mathworld.wolfram.com/MatrixInverse.html
+    """
+    a, b, c, d, e, f = matrix
+
+    determinant = a * d - b * c
```

Comment:
You probably want to cast this to a float. In py2 land if all the matrix entries are ints this'll muck things up.